gateway-messages/src/lib.rs

```diff
@@ -136,6 +146,96 @@ bitflags! {
     }
 }
 
+#[derive(Clone, PartialEq, SerializedSize)]
+pub struct BulkIgnitionState {
+    /// Number of ignition targets present in `targets`.
+    pub num_targets: u16,
+    /// Ignition state for each target.
+    ///
+    /// TODO The ignition target is implicitly the array index; is that
+    /// reasonable or should we specify target indices explicitly?
+    pub targets: [IgnitionState; Self::MAX_IGNITION_TARGETS],
+}
+
+impl fmt::Debug for BulkIgnitionState {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let mut debug = f.debug_struct("BulkIgnitionState");
+        debug.field("num_targets", &self.num_targets);
+        let targets = &self.targets[..usize::from(self.num_targets)];
+        debug.field("targets", &targets);
+        debug.finish()
+    }
+}
+
+impl BulkIgnitionState {
+    // TODO should we set the max to something higher than we expect in rack v1
+    // so we have room to grow without increasing it? Need to think about how
+    // bumping the max affects versioning; higher max doesn't necessarily mean
+    // incompatible messages, but actually sending a packet with a length
+    // greater than the "old" max would fail to parse on an old build.
+    pub const MAX_IGNITION_TARGETS: usize = 36;
```

Comment:
perhaps this should be sized according to how many we can pack into a single UDP message?